### ISSUE TITLE
Is there any dark mode for this app?

### ISSUE THREAD

**Issue by FraH90** [kurema/BookViewerApp3#5]
Hi, first of all this app is the BEST epub reader that I've found for windows.
However, I feel the lack of two functionalities that I still haven't found in this app:

- the dark mode, to read ebooks in inverted colors (black background, white text)
- the possibility to highlight text in the books (obviusly storing them on the epub, after the file is closed)

I would love very very much if those functionalities would be added!

**Comment by kurema** [comment]
Thanks for you suggestion. It sounds to me nice idea to support dark theme for epub viewer.
But we have two problem.

* UWP [WebView](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Controls.WebView?view=winrt-20348) which is based on edgeHTML doesn't seem to support `prefers-color-scheme`.
* Bibi (the engine of epub reader) does not support dark theme yet ([issue](https://github.com/satorumurmur/bibi/issues/93)).

We plan to move to WinUI 3 / WebView2 (based on Chromium). It will fix the first problem. But it should take some time (maybe 2022).
About Bibi, it can be difficult. Epub files have its own css and they don't support dark theme in most cases. It is easy to add 
```css
@media (prefers-color-scheme: dark) { :root{filter: invert(100%) hue-rotate(180deg);}  }
```
but I think it causes numerous problems. Such as
* It does not work correctly if css of the epub file already support dark theme.
* Image can be strange.

In conclusion, we really think it's lovely idea but it should take a time to implement.

**Comment by kurema** [comment]
I noticed some other small issue for the dark theme. They will be fixed in next release.
![image](https://user-images.githubusercontent.com/10556974/129500031-9ddb407c-d6cc-40df-898f-a20c98bd4bfb.png)
![image](https://user-images.githubusercontent.com/10556974/129500043-c58c0f7d-4ce1-45c5-b158-904fd034fca8.png)

**Comment by FraH90** [comment]
> Thanks for you suggestion. It sounds to me nice idea to support dark theme for epub viewer.
> But we have two problem.
> 
> * UWP [WebView](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Controls.WebView?view=winrt-20348) which is based on edgeHTML doesn't seem to support `prefers-color-scheme`.
> * [Bibi](https://github.com/satorumurmur/bibi) (the engine of epub reader) does not support dark theme yet ([issue](https://github.com/satorumurmur/bibi/issues/93)).
> 
> We plan to move to WinUI 3 / WebView2 (based on Chromium). It will fix the first problem. But it should take some time (maybe in 2022 / [WinUI 3 roadmap](https://github.com/microsoft/microsoft-ui-xaml/blob/main/docs/roadmap.md)).
> About Bibi, it can be difficult. Epub files have its own css and they don't support dark theme in most cases. It is easy to add 
> ```css
> @media (prefers-color-scheme: dark) { :root{filter: invert(100%) hue-rotate(180deg);}  }
> ```
> somewhere but I think it causes numerous problems. Such as
> * It does not work correctly if css of the epub file already support dark theme.
> * Image can be strange.
> 
> In conclusion, we really think it's lovely idea but it should take a time to implement.

I perfectly get it..  so it's just matter of time, that you move the platform to WebView2 and wait for bibi to implement the dark mode. 

And what about the other functionality I mentioned? The possibility to highlights some passages in the epub books?

**Comment by kurema** [comment]
Sorry, I forgot to mention that. About highlighting, I think it's also a issue of Bibi. I don't think I can do anything to implement (other than creating a pull request. which I don't.)
There is a [issue](https://github.com/satorumurmur/bibi/issues/63) on Bibi, but I don't think they will implement.
